### ISSUE TITLE
Bug: incomplete list items should not be considered headings (affects various heading-related rules)

### ISSUE THREAD

**Issue by chrisgrieser** [rvben/rumdl#254]
When you start the next list item (`|` being the cursor), you often end up in this situation:

```md
- Apple
- Orange
- |
```

Due to text with only `-` underneath, rumdl treats `- Orange` as a setext heading, and thus reports various related diagnostics, which are all false positive since I am writing a list, not a heading:

<img width="1315" height="376" alt="Image" src="https://github.com/user-attachments/assets/0bac8883-7667-460c-8e8f-5457e6c2a028" />

However, as far as I can tell by checking GitHub preview or pandoc, in markdown spec a line followed by a line with just `-` is not considered a heading. For a line to be considered a setext heading, the line itself may not be a list item, and must also be preceded by a blank.

```md
- Apple

Orange
- |
```

**Comment by h0uter** [comment]
I am also experiencing a problem related to this.

```md
# hello world

- coool
-
```

Rumdl formats that to:

```md
# hello world

## - coool
-
```

Running 

```sh
rumdl fmt test.md 
 /test.md:4:1: [MD003] Heading style should be # Heading, found Heading (underlined with -) [fixed]
```

I would expect the single `-` to just be treated as an empty list item, e.g. no operation or just removing the empty list item.
